test/suites/basic.sh

```diff
@@ -404,7 +404,7 @@ test_basic_usage() {
   lxc launch testimage foo -e
 
   OLD_INIT=$(lxc info foo | grep ^Pid)
-  lxc exec foo reboot
+  lxc restart foo
```

Comment:
So this particular test is specifically for container-initiated reboot, so we should keep it using exec.